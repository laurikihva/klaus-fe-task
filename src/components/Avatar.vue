<template>
  <div class="flex gap-3 items-center">
    <div class="w-[32px] h-[32px] rounded-full overflow-hidden text-xs">
      <img :src="image" :alt="name" />
    </div>
    <div class="flex flex-col gap-0.5 text-sm">
      <span class="font-medium">{{ name }}</span>
      <span class="text-gray-700">{{ email }}</span>
    </div>
  </div>
</template>

<script lang="ts" setup>
export interface IAvatarProps {
  image: string;
  name: string;
  email: string;
}
defineProps<IAvatarProps>();
</script>
