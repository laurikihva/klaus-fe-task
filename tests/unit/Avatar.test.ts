import { mount } from "@vue/test-utils";

import Avatar from "@/components/Avatar.vue";

const wrapper = mount(Avatar, {
  props: {
    name: "Test Name",
    email: "test.name@email.com",
    image: "https://placehold.co/32x32",
  },
});

describe("Loads and displays Avatar component", () => {
  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("contains name, email and base classes", () => {
    expect(wrapper.text()).toContain("Test Name");
    expect(wrapper.text()).toContain("test.name@email.com");

    expect(wrapper.classes("flex")).toBe(true);
    expect(wrapper.classes("gap-3")).toBe(true);
    expect(wrapper.classes("items-center")).toBe(true);
  });
});
