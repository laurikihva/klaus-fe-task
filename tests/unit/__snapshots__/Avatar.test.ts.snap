// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loads and displays Avatar component > renders correctly 1`] = `
"<div class="flex gap-3 items-center overflow-hidden">
  <div class="w-[32px] h-[32px] rounded-full overflow-hidden text-xs shrink-0"><img src="https://placehold.co/32x32" alt="Test Name"></div>
  <div class="flex flex-col gap-0.5 text-sm overflow-hidden"><span class="font-medium text-ellipsis overflow-hidden">Test Name</span><span class="text-gray-700 text-ellipsis overflow-hidden">test.name@email.com</span></div>
</div>"
`;
